### PR TITLE
Commerce 9018

### PR DESCRIPTION
**Pull request by MahmoudAzzam**

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:sf

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ImportTaskResourceImpl.java [hunk 1]
```diff
@@ -419,8 +435,17 @@ private ImportTask _importFile(
 				binaryFile.getInputStream());
 		}
 		else {
+			String fileName = binaryFile.getFileName();
+
+			String fileExtension = _file.getExtension(fileName);
+
+			String trimmedFileName = fileName.replaceAll(
+				"-.*.", "." + fileExtension);
```
Comment (igorbeslic) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ImportTaskResourceImpl.java:443]: this is not good. Please try this with file name:
file-name-with-many-dashes-to-replace-randomXyZ.csv

please instead regex use:
fileName.substring(0, fileName.lastIndexOf("-")) + "." +fileExtension

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ImportTaskResourceImpl.java [hunk 2]
```diff
@@ -343,15 +349,21 @@ private Response _getImportTaskContent(
 				batchEngineImportTask.getBatchEngineImportTaskId()),
 			outputStream);
 
+		String normalizedFileName =
+			(fileName != null) ? _normalizeFileName(fileName) :
```
Comment (igorbeslic) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/ImportTaskResourceImpl.java:353]: ternary operator is not used in Liferay :(
please workout it with `if` statements

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/build.gradle [hunk 3]
```diff
@@ -13,7 +13,9 @@ dependencies {
 	compileOnly group: "org.osgi", name: "org.osgi.service.component.annotations", version: "1.4.0"
 	compileOnly group: "org.osgi", name: "osgi.core", version: "6.0.0"
 	compileOnly project(":apps:batch-engine:batch-engine-api")
+	compileOnly project(":apps:batch-planner:batch-planner-api")
```
Comment (igorbeslic) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/build.gradle:16]: please remove this dependency -

**Review by igorbeslic — CHANGES_REQUESTED** [review]
Hey please fix and resend

**Pull request closed without merging**